include/mxnet/c_api.h

```diff
@@ -548,15 +549,14 @@ MXNET_DLL int MXNDArrayCreate(const mx_uint *shape,
  * \param out the returning handle
  * \return 0 when success, -1 when failure happens
  */
-MXNET_DLL int MXNDArrayCreateEx(const mx_uint *shape,
+MXNET_DLL int MXNDArrayCreateEx(const mx_int64 *shape,
```

Comment:
Sorry if this is a stupid question, but why are we replacing `mx_uint` (unsigned) with `mx_int64` (signed)?
I would assume shape is always positive, shouldn't it be `mx_uint64`?

Reply:
Two reasons:
(1) According to Google C++ style guide: "Do not use an unsigned type merely to assert that a variable is non-negative." https://google.github.io/styleguide/cppguide.html#Integer_Types

(2) We intentionally changed to int because we may reserve the index -1 for supporting empty tensor.

Reply:
Oh. Thank You for the explanation.